framework/src/Volo.Abp.Caching/Volo/Abp/Caching/DistributedCache.cs

```diff
@@ -439,4 +439,327 @@ protected virtual void SetDefaultOptions()
             DefaultCacheOptions = GetDefaultCacheEntryOptions();
         }
     }
+    /// <summary>
+    /// Represents a distributed cache of <typeparamref name="TCacheItem" /> type.
+    /// Uses a generic cache key type of <typeparamref name="TCacheKey" /> type.
+    /// </summary>
+    /// <typeparam name="TCacheItem">The type of cache item being cached.</typeparam>
+    /// <typeparam name="TCacheKey">The type of cache key being used.</typeparam>
+    public class DistributedCache<TCacheItem, TCacheKey> : IDistributedCache<TCacheItem, TCacheKey>
```

Comment:
Duplicating the whole class code is not a good idea, because all the code of these two classes will change together in the future.

Actually `DistributedCache<TCacheItem>` is a special type of `DistributedCache<TCacheItem, TCacheKey>`. So, why not inherit `DistributedCache<TCacheItem>` from `DistributedCache<TCacheItem, string>`? In this way, no code duplication occurs.
Same is possible for the interfaces.